<launch>
    <!-- Set the path to the environment -->
    <arg name = "venv"      default = "$(optenv HRI_CONVERSATIONAL_AGENCY_VENV)" />
    <arg name = "agent_cfg_yaml_path" default = "$(find hri_conversational_agency)/cfg/multiprompt_demo.yaml" />
    <arg name = "ns"        default = "conversational_agent" />
    <arg name = "request_topic"  default = "request"   />
    <arg name = "response_topic" default = "response"  />

    <!-- Launch the node to chat with the conversational agent -->
    <node pkg = "hri_conversational_agency" type = "chat_advanced.py" name = "chat" ns="$(arg ns)"
    output = "screen" launch-prefix="$(arg venv)" >
        <param name = "agent_cfg_yaml_path" value = "$(arg agent_cfg_yaml_path)" />
        <remap from = "request"   to = "$(arg request_topic)" />
        <remap from = "response"  to = "$(arg response_topic)" />
    </node>
</launch>
